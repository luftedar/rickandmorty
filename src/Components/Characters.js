import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Character from './Character';
import { fetchAllCharacters } from '../Redux/characters/characters';
import Page from './Page';

function Characters() {
  const dispatch = useDispatch();
  const characters = useSelector((state) => state.charactersReducer);
  useEffect(() => { dispatch(fetchAllCharacters()); }, []);
  const maxPageValue = Array.from({ length: 42 }, (_, i) => i + 1);
  const newPage = (pageValue) => {
    console.log(pageValue);
    dispatch(fetchAllCharacters(pageValue));
  };
  console.log(characters);
  return (
    <div className="character-container">
      <select name="page" id="pages" onChange={(e) => { newPage(e.target.value); }}>
        {maxPageValue.map((number) => (
          <Page
            key={number}
            number={number}
          />
        ))}
      </select>
      <ul>
        {characters.map((character) => (
          <Character
            key={character.id}
            character={character}
          />
        ))}
      </ul>
    </div>
  );
}

export default Characters;
