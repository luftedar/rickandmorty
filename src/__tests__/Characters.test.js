import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { Provider } from 'react-redux';
import store from '../__mocks__/configureStore';
import Characters from '../Components/Characters';

describe('App page is working fine', () => {
  test('renders App', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Characters />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
