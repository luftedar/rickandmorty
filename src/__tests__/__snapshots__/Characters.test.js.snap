// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App page is working fine renders App 1`] = `
<div
  className="character-container"
>
  <div
    className="home-go-back"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.83 5.146a.5.5 0 0 0 0 .708L7.975 8l-2.147 2.146a.5.5 0 0 0 .707.708l2.147-2.147 2.146 2.147a.5.5 0 0 0 .707-.708L9.39 8l2.146-2.146a.5.5 0 0 0-.707-.708L8.683 7.293 6.536 5.146a.5.5 0 0 0-.707 0z"
        />
        <path
          d="M13.683 1a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2h-7.08a2 2 0 0 1-1.519-.698L.241 8.65a1 1 0 0 1 0-1.302L5.084 1.7A2 2 0 0 1 6.603 1h7.08zm-7.08 1a1 1 0 0 0-.76.35L1 8l4.844 5.65a1 1 0 0 0 .759.35h7.08a1 1 0 0 0 1-1V3a1 1 0 0 0-1-1h-7.08z"
        />
      </svg>
    </a>
  </div>
  <div
    className="character-header"
  >
    Characters
    <br />
    Total 826, Select Page Number From Dropdown List to Navigate
  </div>
  <div
    className="character-listing"
  >
    <div
      className="filter-status d-flex"
    >
      <div>
        Filter By Character
      </div>
      <select
        id="pages"
        name="page"
        onChange={[Function]}
      >
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
        <option
          value={4}
        >
          4
        </option>
        <option
          value={5}
        >
          5
        </option>
        <option
          value={6}
        >
          6
        </option>
        <option
          value={7}
        >
          7
        </option>
        <option
          value={8}
        >
          8
        </option>
        <option
          value={9}
        >
          9
        </option>
        <option
          value={10}
        >
          10
        </option>
        <option
          value={11}
        >
          11
        </option>
        <option
          value={12}
        >
          12
        </option>
        <option
          value={13}
        >
          13
        </option>
        <option
          value={14}
        >
          14
        </option>
        <option
          value={15}
        >
          15
        </option>
        <option
          value={16}
        >
          16
        </option>
        <option
          value={17}
        >
          17
        </option>
        <option
          value={18}
        >
          18
        </option>
        <option
          value={19}
        >
          19
        </option>
        <option
          value={20}
        >
          20
        </option>
        <option
          value={21}
        >
          21
        </option>
        <option
          value={22}
        >
          22
        </option>
        <option
          value={23}
        >
          23
        </option>
        <option
          value={24}
        >
          24
        </option>
        <option
          value={25}
        >
          25
        </option>
        <option
          value={26}
        >
          26
        </option>
        <option
          value={27}
        >
          27
        </option>
        <option
          value={28}
        >
          28
        </option>
        <option
          value={29}
        >
          29
        </option>
        <option
          value={30}
        >
          30
        </option>
        <option
          value={31}
        >
          31
        </option>
        <option
          value={32}
        >
          32
        </option>
        <option
          value={33}
        >
          33
        </option>
        <option
          value={34}
        >
          34
        </option>
        <option
          value={35}
        >
          35
        </option>
        <option
          value={36}
        >
          36
        </option>
        <option
          value={37}
        >
          37
        </option>
        <option
          value={38}
        >
          38
        </option>
        <option
          value={39}
        >
          39
        </option>
        <option
          value={40}
        >
          40
        </option>
        <option
          value={41}
        >
          41
        </option>
        <option
          value={42}
        >
          42
        </option>
      </select>
    </div>
    <ul
      className="list-of-characters"
    />
  </div>
</div>
`;
